src-self-hosted/translate_c.zig

```diff
@@ -426,6 +440,89 @@ fn resolveTypeDef(c: *Context, typedef_decl: *const ZigClangTypedefNameDecl) Err
     try addTopLevelDecl(c, typedef_name, &node.base);
 }
 
+fn resolveRecordDecl(c: *Context, record_decl: *const ZigClangRecordDecl) Error!void {
+    if (try c.decl_table.put(@ptrToInt(ZigClangRecordDecl_getCanonicalDecl(record_decl)), {})) |_| return; // Avoid processing this decl twice
+    const rp = makeRestorePoint(c);
+
+    const bare_name = try c.str(ZigClangDecl_getName_bytes_begin(@ptrCast(*const ZigClangDecl, record_decl)));
+
+    const container_kind_name = if (ZigClangRecordDecl_isUnion(record_decl))
+        "union"
+    else if (ZigClangRecordDecl_isStruct(record_decl))
+        "struct"
+    else
+        return failDecl(c, ZigClangRecordDecl_getLocation(record_decl), bare_name, "record {} is not a struct or union", .{bare_name});
+
+    if (ZigClangRecordDecl_isAnonymousStructOrUnion(record_decl) or bare_name.len == 0)
+        return;
+
+    const visib_tok = try appendToken(c, .Keyword_pub, "pub");
+    const const_tok = try appendToken(c, .Keyword_const, "const");
+
+    const name = try std.fmt.allocPrint(c.a(), "{}_{}", .{ container_kind_name, bare_name });
+    const name_tok = try appendToken(c, .Identifier, name);
+
+    const eq_tok = try appendToken(c, .Equal, "=");
+    const init_node = transRecordDecl(c, record_decl) catch |err| switch (err) {
+        error.UnsupportedType => {
+            return failDecl(c, ZigClangRecordDecl_getLocation(record_decl), name, "unable to resolve record type", .{});
+        },
+        error.OutOfMemory => |e| return e,
+    };
+    const semicolon_token = try appendToken(c, .Semicolon, ";");
+
+    const node = try c.a().create(ast.Node.VarDecl);
+    node.* = ast.Node.VarDecl{
+        .base = ast.Node{ .id = .VarDecl },
+        .doc_comments = null,
+        .visib_token = visib_tok,
+        .thread_local_token = null,
+        .name_token = name_tok,
+        .eq_token = eq_tok,
+        .mut_token = const_tok,
+        .comptime_token = null,
+        .extern_export_token = null,
+        .lib_name = null,
+        .type_node = null,
+        .align_node = null,
+        .section_node = null,
+        .init_node = init_node,
+        .semicolon_token = semicolon_token,
+    };
+
+    try addTopLevelDecl(c, name, &node.base);
+    try c.alias_list.push(.{ .alias = bare_name, .name = name });
+}
+
+fn createAlias(c: *Context, alias: var) !void {
+    const visib_tok = try appendToken(c, .Keyword_pub, "pub");
+    const mut_tok = try appendToken(c, .Keyword_const, "const");
+    const name_tok = try appendToken(c, .Identifier, alias.alias);
+
+    const eq_tok = try appendToken(c, .Equal, "=");
+    const init_node = try appendIdentifier(c, alias.name);
+
+    const node = try c.a().create(ast.Node.VarDecl);
+    node.* = ast.Node.VarDecl{
+        .base = ast.Node{ .id = .VarDecl },
```

Comment:
Btw feel free to make all the ast nodes `base` fields have default values that initialize the id. This will allow omitting the base field at initialization sites, avoiding the possible mistake of having the wrong id. I did this for a few types but not all of them.